src/iperf_api.c

```diff
@@ -1095,7 +1095,8 @@ iperf_parse_arguments(struct iperf_test *test, int argc, char **argv)
         {"udp", no_argument, NULL, 'u'},
         {"bitrate", required_argument, NULL, 'b'},
         {"bandwidth", required_argument, NULL, 'b'},
-	{"server-bitrate-limit", required_argument, NULL, OPT_SERVER_BITRATE_LIMIT},
+        {"server-bitrate-limit", required_argument, NULL, OPT_SERVER_BITRATE_LIMIT},
+        {"server-max-duration", required_argument, NULL, OPT_SERVER_MAX_DURATION},
```

Comment:
There is no need to add a new option, as `--time` can be used (the iperf3 team don't like new options).  Client will store the value in `duration` and server will store it in `max_server_duration`.  Note that when getting the `--time` value it should be stored in temporary variable, and only after all option values are received and it is known whether this is a server or client, it can be stored into the test structure.

This requires that the `--time` will become server and client feature, and the help message should state both the client and the server messages.  Note that

Reply:
The approach I've taken was to be consistent with [iperf/#999](https://github.com/esnet/iperf/pull/999), where the client's `--bitrate` flag and the server's `--server-bitrate-limit` flag are handled separately under different options even though they are both related to configuring bit rate. I'm in favor of maintaining consistency for the sake of the user's experience, but I also think that `--server-max-duration` makes it more clear that this flag has to do with test duration enforcement. We can even change it to `--server-max-test-duration` and `OPT_SERVER_MAX_TEST_DURATION` to be more consistent with the previous recommendation for changing `IEMAXSERVERDURATIONEXCEEDED` to `IEMAXSERVERTESTDURATIONEXCEEDED`. What do you think?

Reply:
I think adding a new flag is OK this case.